app/src/main/java/unipd/se18/ocrcamera/ResultActivity.java

```diff
@@ -299,5 +320,42 @@ public String inciDetectorEtichetta(String text){
         return inci;
     }
 
+
+    /**
+     * @autor Giovanni Fasan(g1)
+     * Function that opens a new activity
+     * https://developer.android.com/reference/android/provider/MediaStore.Images.Media
+     */
+    private void openGallery(){
+        Intent gallery = new Intent(Intent.ACTION_PICK, MediaStore.Images.Media.INTERNAL_CONTENT_URI);      //open device gallery
+        startActivityForResult(gallery, PICK_IMG);  //start gallery's intent
+    }
+
+    /**
+     * @autor Giovanni Fasan(g1)
+     * @param requestCode integer code
+     * @param resultCode integer code
+     * @param data  Intent variable
+     * onActivityResult start another activity and receive a result back
+     * Save the data in a Uri variable. Convert the Uri variable to bitmap to be able to execute the ocr. Set the imageview and execute the ocr.
+     * https://developer.android.com/reference/android/provider/MediaStore.Images.Media
+     */
+    protected void onActivityResult(int requestCode, int resultCode, Intent data){
+        super.onActivityResult(requestCode,resultCode,data);
+        if(resultCode==RESULT_OK && requestCode==PICK_IMG){
+            Uri imageUri;
+            imageUri = data.getData();
+            try {
+                lastPhoto = MediaStore.Images.Media.getBitmap(this.getContentResolver(), imageUri);
+                mImageView = findViewById(R.id.img_captured_view);
+                mImageView.setImageBitmap(Bitmap.createScaledBitmap(lastPhoto, lastPhoto.getWidth(), lastPhoto.getHeight(), false));
+                AsyncLoad ocrTask = new AsyncLoad(mOCRTextView,getString(R.string.processing));
+                ocrTask.execute(lastPhoto);
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+        }
+    }
+
 }
 
```

Comment:
I cannot comment all the line so i made all the other comments here:
-line 254 and 278  Could be better to add a comment to the method that explains what it does.